### DISCUSSION TITLE
Drop Python 3.7 support

### DISCUSSION THREAD

**Discussion by sisp** [copier-org/copier#1251] [Ideas]
[Python 3.7 reached EOL on June 27](https://devguide.python.org/versions/). Should we drop support? It would simplify the code base a bit, e.g.:

* We could remove the [`shutil.copytree` backport](https://github.com/copier-org/copier/blob/98885713eb91d0c4a87984a56b2f47fd67022e35/copier/main.py#L60-L67).
* We could simplify the `typing.{Literal,Protocol,get_args}` imports which require `typing-extensions` for Python 3.7.
* We could simplify the `cached_property` imports which require `backports` for Python 3.7, and we could drop the `backports` dependency.
* We could simplify the [`importlib.metadata` import](https://github.com/copier-org/copier/blob/98885713eb91d0c4a87984a56b2f47fd67022e35/copier/tools.py#L22-L26) and drop the `importlib-metadata` dependency.
* And perhaps more ...

Would this require a new major release? I think projects handle dropping Python version support differently, some consider it a breaking change, some don't.

**Comment by yajo** [comment]
Yes, of course we should do this.

Dependency changes are not breaking changes, according to semver.
